### PR TITLE
Python bump to min 3.8

### LINKED ISSUE #51: Remove Python 3.7 support
As the title says, remove Py 3.7 support.
3.11 should be supported too.

### PR DESCRIPTION
**Pull request by fdosani**
Fixes #51 

- Min Python version is now 3.8
- bumping version
- CODEOWNERS: adding in @NikhilJArora, @jdawang and @gladysteh99

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ repos:
       - id: flake8
         additional_dependencies: [flake8-docstrings]
         files: ^edgetest_pip_tools/
-        language_version: python3.9
+        language_version: python3.10
   - repo: https://github.com/jazzband/pip-tools
     rev: 5.5.0
```
Comment (jdawang) [inline .pre-commit-config.yaml:21]: Should upgrade our pre-commit hook version for pip-tools as it's out of the range in `pyproject.toml`. Does this even do anything with requirements.txt deleted though?

File: .github/workflows/edgetest.yml [hunk 2]
```diff
@@ -14,9 +14,9 @@ jobs:
         with:
           ref: dev
       - id: run-edgetest
-        uses: fdosani/run-edgetest-action@v1.2
+        uses: fdosani/run-edgetest-action@v1.3
         with:
           edgetest-flags: '-c setup.cfg --export'
```
[the remark this answers is not in the document]
Comment (fdosani) [reply .github/workflows/edgetest.yml:19]: Not it updates `setup.cfg`

File: .pre-commit-config.yaml (hunk 1 above)
Comment (fdosani) [reply .pre-commit-config.yaml:21]: Yup good call. I hate having specific versions. Just the latest is fine.

Comment (fdosani) [reply .pre-commit-config.yaml]: Updated the file here with all the latest versions and urls

**Review by jdawang — APPROVED** [review]

**Pull request merged**